transport/src/main/java/io/camunda/zeebe/transport/stream/impl/RemoteStreamTransport.java

```diff
@@ -69,45 +94,109 @@ public void removeAll(final MemberId member) {
     requestHandler.removeAll(sender);
-    return EMPTY_PAYLOAD;
+    return ArrayUtil.EMPTY_BYTE_ARRAY;
   }
 
-  public void recreateStreams(final MemberId receiver) {
+  public CompletableFuture<Void> restartStreams(final MemberId receiver) {
+    final var completed = new CompletableFuture<Void>();
+    sendRestartStreamsRequest(receiver, completed, INITIAL_RETRY_DELAY_MS);
+    return completed;
+  }
+
+  private void sendRestartStreamsRequest(
+      final MemberId receiver, final CompletableFuture<Void> completed, final long retryDelayMs) {
     try {
-      LOG.debug("Restarting streams with for newly added member '{}'", receiver);
-      sendRestartStreamsCommand(receiver)
-          .whenComplete(
-              (ok, error) -> {
-                if (error != null) {
-                  LOG.warn("Failed to restart streams for member '{}'", receiver, error);
-                } else {
-                  LOG.debug("Restarted streams for member '{}'", receiver);
-                }
-              });
+      sendRestartStreamsRequest(receiver)
+          .whenCompleteAsync(
+              (ok, error) -> onRestartStreamsResponse(receiver, error, completed, retryDelayMs),
+              actor);
     } catch (final Exception e) {
       LOG.warn("Failed to restart streams for member '{}'", receiver, e);
     }
   }
 
-  private CompletableFuture<Void> sendRestartStreamsCommand(final MemberId receiver) {
+  private CompletableFuture<Void> sendRestartStreamsRequest(final MemberId receiver) {
     return transport
         .send(
             StreamTopics.RESTART_STREAMS.topic(),
-            EMPTY_PAYLOAD,
+            ArrayUtil.EMPTY_BYTE_ARRAY,
             Function.identity(),
             Function.identity(),
             receiver,
             REQUEST_TIMEOUT)
         .thenApply(ok -> null);
   }
+
+  private void onRestartStreamsResponse(
+      final MemberId receiver,
+      final Throwable error,
+      final CompletableFuture<Void> completed,
+      final long retryDelayMs) {
+    if (error == null) {
+      LOG.debug("Requested streams from client service member '{}'", receiver);
+      completed.complete(null);
+      return;
+    }
+
+    final var cause = error.getCause();
+    switch (cause) {
+        // it's possible that the member that was just added has since been removed in between
+        // retries;
+        // if this is the case, it'll be re-added eventually
+      case final NoSuchMemberException e -> {
+        LOG.trace(
+            """
+            Failed to restart streams for member '{}', which has been removed from the
+            membership protocol; can be safely ignored.""",
+            receiver,
+            e);
+        completed.complete(null);
+      }
+        // this error means the remote member is not handling requests of this type; either it's not
+        // a
+        // gateway, or it's still starting up or shutting down. in either case, restarting streams
+        // makes
+        // no sense
```

Comment:
very small format nitpick

```suggestion
        // this error means the remote member is not handling requests of this type; either it's not
        // a gateway, or it's still starting up or shutting down. in either case, restarting streams
        // makes no sense
```

Reply:
Oops, that's when I pre-emptively put a breakline, then auto format :smile: